lib/grammers-client/src/client/chats.rs

```diff
@@ -848,6 +848,70 @@ impl Client {
             Some(_) => Ok(None),
         }
     }
+
+    /// Send a message action (such as typing, uploading photo, or viewing an emoji interaction)
+    ///
+    /// # Examples
+    ///
+    /// **Do a one-shot pulse and let it fade away**
+    /// ```
+    /// # async fn f(chat: grammers_client::types::Chat, client: grammers_client::Client) -> Result<(), Box<dyn std::error::Error>> {
+    /// use grammers_tl_types::enums::SendMessageAction;
+    ///
+    /// client
+    ///     .action(&chat)
+    ///     .oneshot(SendMessageAction::SendMessageTypingAction)
+    ///     .await?;
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// **Repeat request until the future is done**
+    /// ```
+    /// # use std::time::Duration;
+    ///
+    /// # async fn f(chat: grammers_client::types::Chat, client: grammers_client::Client) -> Result<(), Box<dyn std::error::Error>> {
+    /// use grammers_tl_types as tl;
+    ///
+    /// let heavy_task = async {
+    ///     tokio::time::sleep(Duration::from_secs(10)).await;
+    ///
+    ///     42
+    /// };
+    ///
+    /// tokio::pin!(heavy_task);
+    ///
+    /// let (task_result, _) = client
+    ///     .action(&chat)
+    ///     .repeat(
+    ///         // most clients doesn't actually show progress of an action
+    ///         || tl::types::SendMessageUploadDocumentAction { progress: 0 },
+    ///         Duration::from_secs(4), // repeat request every 4 seconds,
```

Comment:
Does this need to be configurable? I think client are either hardcoded to 5 seconds, or do figure it out from `GetConfig`.